java/serve/src/main/java/io/ray/serve/api/ServeControllerClient.java

```diff
@@ -222,68 +230,21 @@ private void waitForDeploymentHealthy(String name) {
    */
   public synchronized void shutdown() {
     if (Ray.isInitialized() && !shutdown) {
-      ((PyActorHandle) controller).task(PyActorMethod.of("shutdown")).remote();
-      waitForDeploymentsShutdown(60);
-
-      controller.kill();
-
-      long started = System.currentTimeMillis();
-      while (true) {
-        Optional<BaseActorHandle> controllerHandle =
-            Ray.getActor(controllerName, Constants.SERVE_NAMESPACE);
-        if (!controllerHandle.isPresent()) {
-          // actor name is removed
-          break;
-        }
-        long currentTime = System.currentTimeMillis();
-        if (currentTime - started > 5000) {
-          LOGGER.warn(
-              "Waited 5s for Serve to shutdown gracefully but the controller is still not cleaned up. You can ignore this warning if you are shutting down the Ray cluster.");
-          break;
-        }
-      }
-
-      shutdown = true;
-    }
-  }
 
-  /**
-   * Waits for all deployments to be shut down and deleted.
-   *
-   * <p>Throw RayServeException if this doesn't happen before timeoutS.
-   *
-   * @param timeoutS
-   */
-  private void waitForDeploymentsShutdown(long timeoutS) {
-    long start = System.currentTimeMillis();
-    List<DeploymentStatusInfo> deploymentStatuses = null;
-    while (System.currentTimeMillis() - start < timeoutS * 1000) {
-      StatusOverview statusOverview = getServeStatus();
-      if (statusOverview == null
-          || statusOverview.getDeploymentStatuses() == null
-          || statusOverview.getDeploymentStatuses().getDeploymentStatusInfosList() == null
-          || statusOverview.getDeploymentStatuses().getDeploymentStatusInfosList().isEmpty()) {
-        return;
-      }
-      deploymentStatuses = statusOverview.getDeploymentStatuses().getDeploymentStatusInfosList();
-      LOGGER.debug("Waiting for shutdown, {} deployments still alive.", deploymentStatuses.size());
       try {
-        Thread.sleep(CLIENT_POLLING_INTERVAL_S * 1000);
-      } catch (InterruptedException e) {
-      }
-    }
-    List<String> liveNames = new ArrayList<>();
-    if (deploymentStatuses != null) {
-      for (DeploymentStatusInfo status : deploymentStatuses) {
-        liveNames.add(status.getName());
+        ((PyActorHandle) controller)
+            .task(PyActorMethod.of("graceful_shutdown"))
+            .remote()
+            .get(30 * 1000);
```

Comment:
what is 30 * 1000?

Reply:
Replaced "30" with the "timeoutS" parameter.